core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java

```diff
@@ -155,6 +161,17 @@ private AwsCredentialsProvider credentialProcessCredentialsProvider() {
                                          .build();
     }
 
+    private AwsCredentialsProvider roleAndWebIdentityTokenProfileCredentialsProvider() {
+        requireProperties(ProfileProperty.ROLE_ARN, ProfileProperty.WEB_IDENTITY_TOKEN_FILE);
+
+        String roleArn = properties.get(ProfileProperty.ROLE_ARN);
+        String roleSessionName = properties.get(ProfileProperty.ROLE_SESSION_NAME);
+        String webIdentityTokenPath = properties.get(ProfileProperty.WEB_IDENTITY_TOKEN_FILE);
+
+        return WebIdentityCredentialsUtils.factory()
+                                          .create(roleArn, roleSessionName, webIdentityTokenPath);
```

Comment:
Can we use a parameter object, since this is cross-module?

Reply:
What's the benefit here? Why wrap all of these if they are already unwrapped?

Reply:
It makes forward-compatibility a lot easier. It's a cheap price to pay for an annoying experience later if we ever need to add something else.